Base widget

ErnestoRiDo opened the pull request:
Updated boilerplate to include /components which will hold aspects of widget itself.  HelpDeskWidget.tsx will display and call function from chatbotApi.ts in order to connect to server.

ErnestoRiDo committed:
task: added base widget and connected to server

SungRKang commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+  // State to hold the current message input by the user
+  const [message, setMessage] = useState('');
+  // // State to hold the history of chat messages
+  const [chatHistory, setChatHistory] = useState<{ sender: string; text: string }[]>([]);
```
This needs to happen on the server side, not client side

SungRKang commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+    if (message.trim()){
+      try{
+        // Appends user message
+        setChatHistory((prevHistory: any) => [...prevHistory, { sender: 'You', text: message }]);
```
don't use any

Ry-ka replied:
Don't use any

SungRKang commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+        const botResponse = await sendMessageToChatbot(message)
+
+        // Appends bot response
+        setChatHistory((prevHistory: any) => [...prevHistory, { sender: 'Bot', text: botResponse }]);
```
don't use any

Ry-ka replied:
Don't use any

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+import { sendMessageToChatbot } from '../chatbotApi.ts';
+
+// import app.css for styling/centering
+import '../App.css'
```
Create a seperate file named "HelpDeskWidget.css"

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+      catch(error){
+        console.error("Error sending message:" , error)
+      }
+      setMessage('');
```
Remove setMessage('') calls  in the try block and the one above  and add a "finally" block

ErnestoRiDo committed:
Update HelpDeskWidget.tsx

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+        setMessage('');
+      }
+      catch(error){
+        console.error("Error sending message:" , error)
```
Error handling is only console facing. Add customer facing behavior.

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+      onChange={(e) => setMessage(e.target.value)}
+      placeholder="Write a message..."
+      style={{marginTop: '10px', width: '80%'}}
+      />
```
```
<input
  type="text"
  value={message}
  onChange={(e) => setMessage(e.target.value)}
  onKeyPress={(e) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      handleSendMessage();
    }
  }}
  placeholder="Write a message..."
  style={{ marginTop: '10px', width: '80%' }}
/>
```

Ry-ka replied:
Above code adds a listener to the enter key

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+      style={{marginTop: '10px', width: '80%'}}
+      />
+      
+      <button onClick={handlesSentMessage} style={{marginTop: '10px'}}>
```
Don't style in line, move this to a css file

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+  const [chatHistory, setChatHistory] = useState<{ sender: string; text: string }[]>([]);
+
+  // Function to handle message from input box
+  const handlesSentMessage = async () => {
```
Naming? `handleSendMessage` maybe?

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+const HelpDeskWidget: React.FC = () => {
+  // State to hold the current message input by the user
+  const [message, setMessage] = useState('');
+  // // State to hold the history of chat messages
```
Double //

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+  // // State to hold the history of chat messages
+  const [chatHistory, setChatHistory] = useState<{ sender: string; text: string }[]>([]);
+
+  // Function to handle message from input box
```
More descriptive comment please

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+
+  // Function to handle message from input box
+  const handlesSentMessage = async () => {
+    if (message.trim()){
```
You should save this in a variable and send this to the bot instead

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+      <h1>ChatBox</h1>
+      <div style={{ border: '1px solid #ccc', padding: '10px', height: '200px', overflowY: 'scroll' }}>
+        {chatHistory.map((msg,index) => (
+          <div key={index}>
```
index isn't a unique key necessarily. This may cause rerenders when new messages are stored. You should use a uuid to assign a unique key

Ry-ka replied:
Time stamps also work

Ry-ka commented on client/chatbotWidget/src/components/HelpDeskWidget.tsx:
```diff
@@ -0,0 +1,61 @@
+    // Container div with some padding and iterating through chathistory for messages to display
+    <div style={{padding: '10px'}}>
+      <h1>ChatBox</h1>
+      <div style={{ border: '1px solid #ccc', padding: '10px', height: '200px', overflowY: 'scroll' }}>
```
Move styling to a css file

natalie2nguyen committed:
trying different models

chose random forest

working model

comments

commented out uneccesarry code

test for model

test

requirements

Commit:
Preparsed the input

Commit:
Froze requirements.txt

natalie2nguyen committed:
split model into two functions

natalie2nguyen committed:
more changes

natalie2nguyen committed:
fixed tests

natalie2nguyen committed:
added key and reponse excel sheets

natalie2nguyen committed:
used sklearn accuracy_score metric as the requirement for the test to pass.

natalie2nguyen committed:
removed unused library in model.py

ErnestoRiDo committed:
task: reowrked based on comments on PR

ErnestoRiDo committed:
chore: minor fixes on spacing and comments

Ry-ka reviewed (approved):
Make sure to rebase

ErnestoRiDo committed:
rebasing

ErnestoRiDo committed:
rebasing

ErnestoRiDo committed:
rebasing

ErnestoRiDo committed:
chore: added helpdeskwidget.tsx

ErnestoRiDo committed:
initial widget with connection to server

ErnestoRiDo committed:
widget with server connection for input/output
